### PR TITLE
TPCQC added PID hypothesis to PID Task

### PR DESCRIPTION
**Pull request by HorstMa**

### CODE REVIEW

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Looks like `dEdxRatio` is not used anywhere. Is this intended?

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Not sure whey the pid hypothesis is calculated here again. The goal should be to use the one carried out during the reconstruction. So I guess this should b (+ naming conventions):
```suggestion
      const auto pidHypothesis = track.getPID().getID();
      if (pidHypothesis <= o2::track::PID::NIDs) {
```
and remove the unnecessary variables added above.

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: I think this can be simplified:
```suggestion
        auto pidHist = mMapHistCanvas[(track.getCharge() > 0) ? "hdEdxVspHypoPos" : "hdEdxVspHypoNeg"][idEdxType];
          pidHist->SetBinContent(pidHist->GetXaxis()->FindBin(pTPC), pidHist->GetYaxis()->FindBin(dEdxTot[idEdxType]), pidHypothesis + 1);
```

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Naming convention
```suggestion
bool PID::processTrack(const o2::tpc::TrackTPC& track, size_t nTracks)
```

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: ?
```suggestion
  // ===| variables required for cutting and filling |===
```

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Is this used? See below.

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: ?

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Detectors/TPC/qc/src/PID.cxx
Comment (HorstMa) [reply Detectors/TPC/qc/src/PID.cxx]: This does not work, unfortunately, since the histograms in the map are unique_ptr. Either I can make them non-unique ptr or we go with the old method

Comment (wiechula) [reply Detectors/TPC/qc/src/PID.cxx]: Can you try
```suggestion
 auto pidHist = mMapHistCanvas[(track.getCharge() > 0) ? "hdEdxVspHypoPos" : "hdEdxVspHypoNeg"][idEdxType].get();
          pidHist->SetBinContent(pidHist->GetXaxis()->FindBin(pTPC), pidHist->GetYaxis()->FindBin(dEdxTot[idEdxType]), pidHypothesis + 1);
```

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Not used any longer.
```suggestion
```

File: Detectors/TPC/qc/src/PID.cxx
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]:
```suggestion
```

File: Detectors/TPC/qc/include/TPCQC/PID.h
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]: Naming convention
```suggestion
  bool processTrack(const o2::tpc::TrackTPC& track, size_t nTracks);
```

File: Detectors/TPC/qc/include/TPCQC/PID.h
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]: As written before, the unneeded members should be removed.
```suggestion
```

**Review by wiechula — CHANGES_REQUESTED** [review]

**Review by wiechula — DISMISSED** [review]

**Review by wiechula — DISMISSED** [review]

**Review by shahor02 — DISMISSED** [review]
Conditionally approving 1st-time submitter to start CI

**Comment by shahor02** [comment]
Hi @ktf , after my approval of 1st time submitter commit it still says `Pending — Security: approval needed, not starting `. 
Usually it worked, did you change something?

**Comment by ktf** [comment]
I did not, and I actually had the same issue on a different PR. @TimoWilken any idea?

**Comment by TimoWilken** [comment]
Hi @shahor02 @ktf, sorry, that status doesn't get cleared immediately! Whenever the builders look for PRs to test the next time, they will pick this PR up. Some of them already say "Started Thu ...".

**Comment by lauraser** [comment]
@wiechula @HorstMa There seems to be an error in the O2 build but it comes from QC file. Could you please check it?

**Comment by shahor02** [comment]
@HorstMa @lauraser Since the signature of the `processTrack(track)` was changed to `processTrack(track, ntracks)`, the QC rightfully complains that it does not find the method. 
One should either leave temporarily a dummy `processTrack(track) {return true;}` method (which can be removed once the QC is modified to use the new method) or to make the new signature compatible with the old one by adding a default value as e.g.  `processTrack(track, ntracks=0)` (if it makes sense).

**Comment by HorstMa** [comment]
@shahor02 @wiechula I added the dummy method. Please reapprove.

**Review by shahor02 — APPROVED** [review]

**Review by wiechula — APPROVED** [review]

**Pull request merged**
